best-practice-certificate-provisioning.md

```diff
@@ -0,0 +1,317 @@
+
+### Root Certificate Authority Renewal
+
+Renewal of the Root CA SHOULD be attempted no sooner than 50% of the certificates expiry time or before the 'Not Before' date on the certificate. It is RECOMMENDED that certificate renewal is performed after 80% of the expiry time. To renew the Root CA and the EST Clients TLS Certificate follow the [Initial Certificate Provisioning](#initial-certificate-provisioning) workflow, using the existing TLS Certificate for authentication if still valid.
+If the returned Root Certificate Authority by the EST Server is the same as the existing Root Certificate Authority, the EST Client should attempt renewal of the Root Certificate Authority after half of the remaining period of validity has elapsed.
+
+### Expired Manufacturer Issued TLS Client Certificate
+
+If the Manufacturer issued TLS Client Certificate has expired or has been revoked, it MUST NOT be used by the EST Client for authentication. An EST Client MAY attempt to request a TLS Certificate following [Initial Certificate Provisioning](#initial-certificate-provisioning), without providing a TLS Client Certificate during the TLS handshake. If the EST Server supports manual authentication the request will be processed.
+
+If the EST Server fails to process the request the following actions MAY be taken:
+1. The EST Client MAY have a TLS Certificate for the target network manually installed on the device. The manually installed TLS certificate MUST then be used to secure its NMOS APIs until the TLS Certificate is due for renewal. The manually installed TLS Certificate MUST be presented used during the TLS Handshake for [Certificate Renewals](#certificate-renewal). If the certificate renewal is successful, the manually installed TLS Certificate and private key MUST be removed.
+2. The Manufacturer issued TLS Client Certificate MAY be renewed by a software/firmware update, this update MUST contain a unique TLS certificate per device. The valid TLS Client Certificate MAY then be used during [Initial Certificate Provisioning](#initial-certificate-provisioning)
+
+### Connection to new network
+
+It MAY be desirable that when a EST Client is connected to a different network that it automatically requests a TLS Certificate for the new network.
+
+On start up or on change of network state the EST Client MUST attempt to discover the EST Server using [DNS-SD](#dns-sd-advertisement), unless manually configured. The EST Client SHOULD make a request to the `/cacerts` endpoint, if the request is successful the EST Client should compare the returned Certificate to the currently install Root CA, if the Certificate is for a different Domain the EST Client MUST follow [Initial Certificate Provisioning](#initial-certificate-provisioning) workflow.
```

Comment:
```suggestion
On start up or on change of network state the EST Client MUST attempt to discover the EST Server using [DNS-SD](#dns-sd-advertisement), unless manually configured. The EST Client SHOULD make a request to the `/cacerts` endpoint, if the request is successful the EST Client SHOULD compare the returned Certificate to the currently installed Root CA. If the Certificate is for a different Domain the EST Client MUST follow the [Initial Certificate Provisioning](#initial-certificate-provisioning) workflow.
```

Reply:
Should we add that devices SHOULD be configurable to disable this domain-changing behaviour?

Reply:
Possibly, but would mean that you might get into a situation where some devices switch when plugged into a new OB truck for example and some don't. 
The risk if domain-swtiching is enabled that you lose control of the device if a rouge EST Server is discovered by the device on the network and control of it is lost